bazaarvoice/emodb
# fix restricted scan behavior

## What Are We Doing Here?

Previously, scanning with an API key could cause you to prematurely
believe the scan was finished. If you got a page for which all results
got filtered out (due to restricted permissions), you would
get back an empty page, which is also the signal for "end-of-scan".

Now, the server will keep grabbing pages until it either hits the end
of the scan or gets enough results post-filter to satisfy your limit.

## How to Test and Verify

1. Check out this PR
2. Run Emo
3. List tables with an api key and see that it works!

## Risk

### Level 

`Low`: this has been broken for quite some time, and I seem to be the only person to have noticed.

### Required Testing

`Smoke`, `Regression`, or `Manual`. (All changes except documentation need smoke
testing at a minimum).

### Risk Summary

Add one or a few complete sentences about the possible risks or concerns for
this change.

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
